File: vol1/ch5/LeetCode49.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string,vector<string>> hmap;
        for(string x:strs) {
            string key=x;
            sort(key.begin(),key.end());
            hmap[key].push_back(x);
        }
        vector<vector<string>> ans;         //´æ·Å´ð°¸
        for(auto x:hmap) {
            ans.push_back(x.second);
        }
        return ans;
    }
};

File: vol1/ch5/LeetCode49.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        hmap=dict()
        for x in strs:
            key="".join((lambda x:(x.sort(),x)[1])(list(x)))
            if key in hmap:
                hmap[key].append(x)
            else:
                hmap[key]=[x]
        ans=[]                  #存放答案
        for v in hmap.values():
            ans.append(v)
        return ans
